docs/K8SCollectors.md

```diff
@@ -0,0 +1,179 @@
+- Root access on the host file system to create/delete directories(atleast within the directory where you want to run this script and store logs)
+- Enough space to collect data in a target folder
+
+### **Script run instructions**
+- Run the script using the following command - `sh k8s-splunk-collector.sh -d <flag_to_collect_splunk_diags> -t <target_folder> -l <flag_to_limit_output_by_avoid_kubectl_describe> -s <collect_secret_data>`. There are three options which are configurable through the script:
+    - The `-d` option is used to specify whether Splunk diags needs to be collected. Splunk diags are not collected by default. Set to `true` if Splunk diags are to be collected.
+    - The `-t` option is used to specify a `<target folder>`. This option is `not mandatory`. The script allows you to store the data collected in two different ways:
+        - If the `-t` option is not used, a timestamped folder `tmp-<timestamp>` is created in the present working directory where the data will be written to. 
+            ```
+            Eg. sh k8s-splunk-collector.sh -d true
+            ```                
+        - If the `-t` option is used with valid full path, a timestamped folder `tmp-<timestamp>` is created inside the full path where the data will be written to. Note: If the folder provided doesn't exist, it is created provided the preceeding path exists i.e in the example below if `/Users/randomuser` is present, `/collect` is created. But if the preceeding path `/Users/randomuser` doesn't exist the script runs into unexpected behavior.
+            ```    
+            Eg. sh k8s-splunk-collector.sh -d true -t /Users/randomuser/collect/
+            ```
+       Please make sure you have enough space in the target folders in either case(for reference look at [performance requirements section](#performance).
+    - The `-l` option is used to specify whether you want to limit the collection of data by avoiding `kubectl describe` commands. The kubectl describe command outputs `are` collected by default. There is an issue in K8S with creating too many clients for describe commands(https://github.com/kubernetes/kubernetes/issues/91913). In internal testing these messages have not caused any issues. However, to avoid the warning messages as well to protect your network bandwidth if limited, you can set the `-l` option to `true`. Example of a warning message from the K8S cluster:
+
+            W0419 14:46:10.239590   21927 exec.go:203] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 1478 clients constructed calling "aws-iam-authenticator"
+    - The `-s` option is used to specify whether K8S secret data needs to be collected. Secret data is not collected by default. Set to `true` if you want the script to collect secret data.
+
+- Wait till you see the message `All data required collected under folder <target_folder>`
```

Comment:
There are a lot of sub-bullets between line 20 and this line. It might help users to write line 37 as "After you run the script, wait until you see the message 'All data required collected under folder <target folder>`"